app/main/posts/modify/post-media.directive.js

```diff
@@ -17,84 +19,44 @@ function (
         require: '^^form',
         scope: {
             mediaId: '=',
+            media: '=',
             name: '@'
         },
         templateUrl: 'templates/main/posts/modify/media.html',
         link: function ($scope, element, attr, formCtrl) {
-            // Initialize file container
-            $scope.fileContainer = {file: null};
+            $scope.showAdd = function () {
+                return (!$scope.media.id && !$scope.media.changed);
+            };
 
-            // Initialize media object
-            $scope.media = {};
+            $scope.showReplace = function () {
+                return $scope.media.dataURI || $scope.media.id;
+            };
+
+            $scope.showDelete = function () {
+                return $scope.media.id;
+            };
 
             if ($scope.mediaId) {
                 MediaEndpoint.get({id: $scope.mediaId}).$promise.then(function (media) {
                     $scope.media = media;
```

Comment:
So the flow is as follows:

1. Retrieve all media attributes and associated them with their key
2. For each media attribute create a media-value directive instances, pass in the media map reference
3. Update/construct object map to medias
4. On save, loop through medias items, taking appropriate action if changed

At the moment, it functions correctly, passing in the media map follows the same pattern as for the other post edit values. 

Could you elucidate why or when you think it'll break? I'm not sure I follow.